<a href="https://colab.research.google.com/github/angelvaz20/Colab_files/blob/main/Xtreme_Boosting_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
#Dillan Andrews, Angel Vasquez
#https://www.kaggle.com/datasets/shaistashahid/gaming-and-mental-health

In [ ]:
#Import Libraries
import xgboost as xgb  # XGBoost library
from sklearn.metrics import accuracy_score, classification_report
from sklearn.model_selection import train_test_split  # For splitting data
from sklearn.metrics import mean_squared_error, mean_absolute_error, r2_score  # Regression metrics
from sklearn.preprocessing import LabelEncoder,StandardScaler
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [ ]:
#import dataset
df = pd.read_csv('Gaming and Mental Health.csv')
df.head()

,record_id,age,gender,daily_gaming_hours,game_genre,primary_game,gaming_platform,sleep_hours,sleep_quality,sleep_disruption_frequency,...,continued_despite_problems,eye_strain,back_neck_pain,weight_change_kg,exercise_hours_weekly,social_isolation_score,face_to_face_social_hours_weekly,monthly_game_spending_usd,years_gaming,gaming_addiction_risk_level
0,GD0001,17,Male,11.1,Mobile Games,Clash of Clans,PC,3.7,Very Poor,Sometimes,...,True,True,False,6.8,3.7,7,1.3,383.70,3,Severe
1,GD0002,21,Male,3.0,MOBA,Dota 2,PC,7.2,Fair,Rarely,...,False,False,False,0.4,8.5,2,10.7,46.64,1,Low
2,GD0003,23,Male,7.6,FPS,CS:GO,Multi-platform,4.4,Fair,Often,...,True,False,True,1.8,7.1,5,3.2,100.81,6,Severe
3,GD0004,20,Female,7.2,RPG,Skyrim,Multi-platform,5.1,Fair,Often,...,False,True,True,0.2,5.2,4,9.1,51.60,7,High
4,GD0005,18,Male,6.8,Battle Royale,Apex Legends,PC,3.4,Poor,Never,...,False,False,False,0.5,6.1,4,4.5,32.57,1,Moderate


In [ ]:
#drop unecessary features
df = df.drop(columns=['record_id','primary_game','withdrawal_symptoms','loss_of_other_interests','continued_despite_problems','eye_strain','back_neck_pain'])
df.dropna(inplace=True)

#Turn categorical data into numerical
le = LabelEncoder()
categorical = ['gender','game_genre','gaming_platform','sleep_quality','sleep_disruption_frequency','academic_work_performance','mood_state','mood_swing_frequency','gaming_addiction_risk_level']
for n in categorical:
  df[n] = le.fit_transform(df[n])
df.head()

,age,gender,daily_gaming_hours,game_genre,gaming_platform,sleep_hours,sleep_quality,sleep_disruption_frequency,academic_work_performance,grades_gpa,work_productivity_score,mood_state,mood_swing_frequency,weight_change_kg,exercise_hours_weekly,social_isolation_score,face_to_face_social_hours_weekly,monthly_game_spending_usd,years_gaming,gaming_addiction_risk_level
1,21,1,3.0,3,3,7.2,0,3,4,3.75,2.0,5,2,0.4,8.5,2,10.7,46.64,1,1
3,20,0,7.2,5,2,5.1,0,2,5,1.62,2.0,8,0,0.2,5.2,4,9.1,51.60,7,0
5,21,1,2.1,4,3,7.5,0,3,4,3.81,4.0,6,1,0.2,10.8,1,9.6,8.63,4,1
6,21,1,2.7,1,2,7.2,4,2,1,1.82,6.0,3,3,0.8,10.0,2,10.1,35.25,3,1
13,22,1,5.1,5,1,5.3,0,1,0,3.25,6.0,1,3,0.9,7.6,3,6.3,36.07,5,1


In [ ]:
#standard scale used features
X = df.drop(columns='gaming_addiction_risk_level')
y = df['gaming_addiction_risk_level']
#scaler = StandardScaler()
#X = scaler.fit_transform(X)
#X = pd.DataFrame(X,columns=['age','gender','daily_gaming_hours','game_genre','gaming_platform','sleep_hours','sleep_quality','sleep_disruption_frequency','academic_work_performance','grades_gpa','work_productivity_score','mood_state','mood_swing_frequency','weight_change_kg','exercise_hours_weekly','social_isolation_score','face_to_face_social_hours_weekly','monthly_game_spending_usd','years_gaming'])
X.head()

,age,gender,daily_gaming_hours,game_genre,gaming_platform,sleep_hours,sleep_quality,sleep_disruption_frequency,academic_work_performance,grades_gpa,work_productivity_score,mood_state,mood_swing_frequency,weight_change_kg,exercise_hours_weekly,social_isolation_score,face_to_face_social_hours_weekly,monthly_game_spending_usd,years_gaming
1,21,1,3.0,3,3,7.2,0,3,4,3.75,2.0,5,2,0.4,8.5,2,10.7,46.64,1
3,20,0,7.2,5,2,5.1,0,2,5,1.62,2.0,8,0,0.2,5.2,4,9.1,51.60,7
5,21,1,2.1,4,3,7.5,0,3,4,3.81,4.0,6,1,0.2,10.8,1,9.6,8.63,4
6,21,1,2.7,1,2,7.2,4,2,1,1.82,6.0,3,3,0.8,10.0,2,10.1,35.25,3
13,22,1,5.1,5,1,5.3,0,1,0,3.25,6.0,1,3,0.9,7.6,3,6.3,36.07,5


In [ ]:
#print(f"Total samples: {len(df)}")
print(f"Number of features: {len(X.columns)}")
print(f"\nFeature names:")
for i, name in enumerate(X, 1):
    print(f"  {i}. {name}")

Number of features: 19

Feature names:
  1. age
  2. gender
  3. daily_gaming_hours
  4. game_genre
  5. gaming_platform
  6. sleep_hours
  7. sleep_quality
  8. sleep_disruption_frequency
  9. academic_work_performance
  10. grades_gpa
  11. work_productivity_score
  12. mood_state
  13. mood_swing_frequency
  14. weight_change_kg
  15. exercise_hours_weekly
  16. social_isolation_score
  17. face_to_face_social_hours_weekly
  18. monthly_game_spending_usd
  19. years_gaming


In [ ]:
# Split: 80% training, 20% testing
X_train, X_test, y_train, y_test = train_test_split(
    X, y,
    test_size=0.2,      # 20% for testing
    random_state=42     # For reproducibility
)

print(f"\nTraining set size: {len(X_train)}")
print(f"Testing set size: {len(X_test)}")


Training set size: 342
Testing set size: 86


In [ ]:
print("\n" + "=" * 70)
print("TRAINING XGBOOST REGRESSOR")
print("=" * 70)

# Initialize XGBoost Regressor
# Note: We use XGBRegressor (not Classifier) for predicting continuous values
model = xgb.XGBRegressor(
    max_depth=7,           # Maximum depth of trees
    n_estimators=100,      # Number of boosting rounds (trees)
    learning_rate=0.1,     # Step size for updating predictions
    objective='reg:squarederror',  # Minimize squared error (standard for regression)
    random_state=42,       # For reproducibility
    subsample=0.8,         # Use 80% of data for each tree (prevents overfitting)
    colsample_bytree=0.8   # Use 80% of features for each tree
)

print("Model parameters:")
print(f"  - Max tree depth: {model.max_depth}")
print(f"  - Number of trees: {model.n_estimators}")
print(f"  - Learning rate: {model.learning_rate}")
print(f"  - Subsample ratio: {model.subsample}")

# Train the model
print("\nTraining in progress...")
model.fit(X_train, y_train, verbose=False)  # verbose=False to reduce output
print("✓ Training complete!")


TRAINING XGBOOST REGRESSOR
Model parameters:
  - Max tree depth: 7
  - Number of trees: 100
  - Learning rate: 0.1
  - Subsample ratio: 0.8

Training in progress...
✓ Training complete!


In [ ]:
# Predict on both training and test sets
y_train_pred = model.predict(X_train)
y_test_pred = model.predict(X_test)
num = 0
for n in y_test_pred:
  n = round(n)
  y_test_pred[num] = n
  num += 1

# Get probability predictions (useful for understanding model confidence)
print('Model Accuracy:',accuracy_score(y_test,y_test_pred))


#work on model to make it more accurate

Model Accuracy: 0.5930232558139535


In [ ]:
print(classification_report(y_test,y_test_pred))

              precision    recall  f1-score   support

           0       0.50      0.05      0.09        21
           1       0.67      0.95      0.78        40
           2       0.48      0.65      0.55        17
           3       0.25      0.12      0.17         8

    accuracy                           0.59        86
   macro avg       0.47      0.44      0.40        86
weighted avg       0.55      0.59      0.51        86



The code above uses an Xtreme Boost algorithm to predict whether a person playing games can be considered to have low, medium, high, or severe risk of gaming addiction. With an algorithm depth of 7, we can get the highest model accuracy of 59.30%. While its accuracy is not high, I believe that it could be improved with a larget data set to train on. With the current training set only being 342, the algorithm has a limited number of people to train on. If a larger dataset with the same features was used, this algorithm could be made more accurate.